lib/galaxy/tools/data/__init__.py

```diff
@@ -610,9 +610,21 @@ def get_filename_for_source(self, source, default=None):
         filename = default
         for name, value in self.filenames.items():
             repo_info = value.get('tool_shed_repository', None)
-            if (not source_repo_info and not repo_info) or (source_repo_info and repo_info and source_repo_info == repo_info):
+            if source_repo_info and repo_info and source_repo_info == repo_info:
                 filename = name
                 break
+        # No toolshed data table entry was found, try the first file found in the OS
+        else:
+            for filename in self.filenames.keys():
+                if os.path.exists(filename):
+                    fmt_dict = dict(source_repo_info)
```

Comment:
`source_repo_info` could be None, so that would fail with
```
>>> dict(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'NoneType' object is not iterable
```
Maybe change the logging below to only log if `source_repo_info` is not None and log something else if it is None ?